keps/sig-scheduling/5075-dra-consumable-capacity/README.md

```diff
@@ -0,0 +1,1535 @@
+   SharingPolicy *CapacitySharingPolicy
+}
+
+// CapacitySharingPolicy defines how requests consume the available capacity.
+// A policy must have a default value to be applied when no value is explicitly provided.
+// It can either specify a range of valid values or a discrete set of them.
+// Exactly one of them must be defined.
+// The default value must be a valid value.
+type CapacitySharingPolicy struct {
+   // Default specifies the default capacity to be used for a consumption request.
+   //
+   // +required
+   Default resource.Quantity
+
+   // DiscreteValues defines a set of acceptable quantity values in consuming requests.
+   //
+   // +optional
+   // +oneOf=ValidSharingValues
+   DiscreteValues *CapacitySharingPolicyDiscrete
+
+   // ValueRange defines an acceptable quantity value range in consuming requests.
+   //
+   // +optional
+   // +oneOf=ValidSharingValues
+   ValueRange *CapacitySharingPolicyRange
+
+   // Potential extension: allow defining a `strategy` on a specific capacity
+   // to specify default scheduling behavior when it is not explicitly requested.
+}
+
+// CapacitySharingPolicyDiscrete defines a set of discrete allowed capacity values.
+//
+// If the requested amount is not listed in the options, it cannot be allocated to this device.
+type CapacitySharingPolicyDiscrete struct {
+    // Options defines a list of additional valid capacity values that can be requested.
+    //
+    // This list must contain only unique items and must include the default value.
+    // All options must be less than or equal to the capacity value.
+    //
+    // The list is limited to at most 64 entries.
+    //
+    // +optional
+    // +listType=atomic
+    Options []resource.Quantity
+}
+
+// CapacitySharingPolicyRange defines a valid range for consumable capacity values.
+//
+// - If the requested amount is less than Minimum, it is rounded up to the Minimum value.
+// - If the requested amount is between Minimum and Maximum, ChunkSize is set,
+//   and the amount does not align with the ChunkSize,
+//   it will be rounded up to the next value matching Minimum + (n * ChunkSize).
+// - If the requested or rounded amount exceeds Maximum (if set), the request does not satisfy the policy,
+//   and the device cannot be allocated.
+//
+// - If ChunkSize is not set, the requested amount is used as-is, provided it falls within the range.
+type CapacitySharingPolicyRange struct {
+    // Minimum specifies the minimum capacity allowed for a consumption request.
+    //
+    // Minimum must be less than or equal to the capacity value.
+    // Default must be more than or equal to the minimum.
+    //
+    // +required
+    Minimum resource.Quantity
+
+    // Maximum defines the upper limit for capacity that can be requested.
+    //
+    // Maximum must be less than or equal to the capacity value.
+    // Minimum and default must be less than or equal to the maximum.
+    //
+    // +optional
+    Maximum *resource.Quantity
```

Comment:
non-blocking, but make a note to decide whether to use Minimum/Maximum or match LimitRange and use Min/Max